pkg/sdn/plugin/node.go

```diff
@@ -127,11 +139,138 @@ func (node *OsdnNode) Start() error {
 		}
 	}
 
+	if err := node.cniServer.Start(); err != nil {
+		return err
+	}
+	go utilwait.Forever(node.watchCni, 0)
+
 	node.markPodNetworkReady()
 
 	return nil
 }
 
+type podWorker struct {
+	node     *OsdnNode
+	requests []*podRequest
+	lock     sync.Mutex
+}
+
+func newPodWorker(node *OsdnNode) *podWorker {
+	return &podWorker{
+		node:     node,
+		requests: make([]*podRequest, 0),
+	}
+}
+
+func (w *podWorker) addRequest(request *podRequest) {
+	w.lock.Lock()
+	defer w.lock.Unlock()
+	w.requests = append(w.requests, request)
+	if len(w.requests) == 1 {
+		// Start proessing when the first request is added
+		go w.process()
+	}
+}
+
+func (w *podWorker) numRequests() int {
+	w.lock.Lock()
+	defer w.lock.Unlock()
+	return len(w.requests)
+}
+
+func (w *podWorker) process() {
+	for {
+		// Grab an item off the queue
+		w.lock.Lock()
+		if len(w.requests) == 0 {
+			w.lock.Unlock()
+			return
+		}
+		request := w.requests[0]
+		w.requests = w.requests[1:]
+		w.lock.Unlock()
```

Comment:
Yeah, it kinda reinvents channels, but with two important differences: (1) it's non-blocking and we don't have to specify a channel capacity (after which things would block) and (2) we can cleanly terminate the pod worker when it no longer has work to do which turns out to be pretty racy with a channel.  (1) is not much of a problem (though annoying) and (2) is harder to solve, but I'll see if I can come up with something not-ugly.